Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/ui/shared/DccTicketingSharedViewModel.kt

```diff
@@ -3,45 +3,51 @@ package de.rki.coronawarnapp.dccticketing.ui.shared
 import androidx.annotation.VisibleForTesting
 import androidx.lifecycle.SavedStateHandle
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import de.rki.coronawarnapp.dccticketing.core.transaction.DccTicketingTransactionContext
+import de.rki.coronawarnapp.tag
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.catch
 import kotlinx.coroutines.flow.filterNotNull
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import timber.log.Timber
 
 /**
  * Shares the [DccTicketingTransactionContext] between all DccTicketing destinations.
  */
 class DccTicketingSharedViewModel(private val savedState: SavedStateHandle) : ViewModel() {
 
-    private val currentTransactionContext = MutableStateFlow(
-        getSavedTransactionContext()
-    )
+    private val currentTransactionContext = MutableStateFlow(savedTransactionContext)
 
     val transactionContext: Flow<DccTicketingTransactionContext> = currentTransactionContext.filterNotNull()
 
+    init {
+        transactionContext
+            .onEach { savedTransactionContext = it }
+            .catch { Timber.tag(TAG).e(it, "Failed to save DccTicketingTransactionContext") }
+            .launchIn(viewModelScope)
+    }
+
     fun updateTransactionContext(ctx: DccTicketingTransactionContext) {
-        Timber.d("updateTransactionContext(ctx=%s)", ctx)
+        Timber.tag(TAG).d("updateTransactionContext(ctx=%s)", ctx)
         currentTransactionContext.tryEmit(ctx).also {
-            Timber.d("Update was successful? %s", it)
-            saveTransactionContext(ctx)
+            Timber.tag(TAG).d("Update was successful? %s", it)
```

Comment:
Please, help me understand why the previous code was problematic ? how this fixes it. `tryEmit` for this state flow always returns `true` 
```kotlin
    override fun tryEmit(value: T): Boolean {
        this.value = value
        return true
    }
```
and despite that previous code was saving the `context` anyway regardless of the output

would not it be just easier if we just use `stateFlow.value = ctx`?

Reply:
The code was not really thread-safe and this could have lead to an inconsistent state, where the current context and the saved context could have differed.
I reworked updating the context to make it fully thread-safe. better safe than sorry

Reply:
Hmm, this is getting a bit complicated. What about getting rid of the `StateFlow` and use `LiveData` instead 🤔.

`SavedStateHandle` can provide LiveData: `savedState.getLiveData<DccTicketingTransactionContext>(TRANSACTION_CONTEXT_SAVED_STATE_KEY)` so after saving something to the `savedState`, the changes are then directly delivered via `LiveData`...

Reply:
That's not really complicated.. I don't see much reason using a LiveData in this case. The transaction context is only called in other ViewModels and never directly exposed to any ui elements without modification. We also would lose the thread safety.
I know that my approach semms a little bit over engineered for the current case. But this could easily change in the furture like when we adding a cancellation service. So like i said, better safe than sorry